ecosystem/attestation-station/using-sdk/README.md

```diff
@@ -0,0 +1,234 @@
+
+If you are going to use `atst` in client-side code to write attestations, use a different connector, probably through [RainbowKit](https://www.rainbowkit.com/docs/installation#configure) of [ConnectKit](https://docs.family.co/connectkit).
+
+```js
+const writeSetup = async () => {
+
+  // Connect the client to a mock connector
+  await wagmiCore.connect({
+    // MockConnector is used for server processes when there is no
+    // user wallet
+    connector: new wagmiCoreMock.MockConnector({
+      options: {
+        chainId: chains[0].ids,
+        signer: new ethers.Wallet(process.env.PRIVATE_KEY, 
+          provider(wagmiChains.optimismGoerli)
+```
+
+A wagmi provider is a function, what gives you [an Ethers `Provider`](https://docs.ethers.org/v5/api/providers/provider/) when you give it the chain you want. 
+
+```js          
+        ),
+      },
+    }),   // end of new wagmiCoreMock.MockConnector
+  })   // end of wagmiCore.connect
+
+}  // end of writeSetup
+
+//  /\  Definitions only required for writing attestations /\
+
+
+const main = async () => {
+
+    setup()
+    await writeSetup()
+```
+
+To write an attestation we need both setups.
+
+```js
+    
+    const readCreatorAddr = "0xc2dfa7205088179a8644b9fdcecd6d9bed854cfe"
+    const aboutAddr = "0x00000000000000000000000000000000000060A7"
+    const key = "animalfarm.school.GPA"
+
+    // Read an attestation
+    const val = await atst.readAttestation(
+        readCreatorAddr,
+        aboutAddr,
+        key,
+        "string")    // data type
+```
+
+Read an attesation.
+The SDK lets you specify what data type you're expecting, and converts the value to that type for you.
+The supported data types are:
+
+- `string` (default)
+- `bytes` 
+- `number`, which is an integer
+- `bool`
+- `address`
+
+You can also specify as an additional paramter the contract address if you are not using the default.
+
+```js
+    
+    console.log(`According to ${readCreatorAddr} the ${key} for ${aboutAddr} is ${val}`)
+    
+    console.log(`--------------`)
+```
+
+```js     
+    const preparedTx = await atst.prepareWriteAttestation(
+      "0x00000000000000000000000000000000000060A7",  // about
+      "animalfarm.school.GPA",                       // key
+      "3.25",                                        // value
+    )
+```
+
+The way wagmi works, you first [prepare a transaction](https://wagmi.sh/react/prepare-hooks) and then send it.
+One advantage of this approach is that the prepared transaction already includes, for example, the gas cost of the transaction (although note that on Optimism that is the tiny [L2 execution fee](https://community.optimism.io/docs/developers/build/transaction-fees/#the-l2-execution-fee), not the much larger [L1 data fee](https://community.optimism.io/docs/developers/build/transaction-fees/#the-l1-data-fee)).
```

Comment:
Another advantage is Prepare Hooks allow you to eagerly perform this asynchronous work beforehand so the user action happens much faster